routing/router.go

```diff
@@ -111,8 +114,8 @@ type Config struct {
 	// forward a fully encoded payment to the first hop in the route
 	// denoted by its public key. A non-nil error is to be returned if the
 	// payment was unsuccessful.
-	SendToSwitch func(firstHop *btcec.PublicKey,
-		htlcAdd *lnwire.UpdateAddHTLC) ([32]byte, error)
+	SendToSwitch func(firstHop *btcec.PublicKey, htlcAdd *lnwire.UpdateAddHTLC,
```

Comment:
I moved the creation of the `htlcswitch.Deobfuscator` to the `SendToSwitch` handler, now it takes the `sphinx.OnionDeobfuscator` as input, and now seems like layer violation have been avoided, at least in terms of importing the `htlcswitch` package.